asserts/database.go

```diff
@@ -219,32 +219,33 @@ func (db *Database) Check(assert Assertion) error {
 		return err
 	}
 	// TODO: later may need to consider type of assert to find candidate keys
-	accKeys, err := db.findAccountKeys(assert.AuthorityID(), sig.KeyID())
+	accKey, err := db.findAccountKey(assert.AuthorityID(), sig.KeyID())
+	if err == ErrNotFound {
+		return fmt.Errorf("no matching public key for signature by %q key id %q", assert.AuthorityID(), sig.KeyID())
+	}
 	if err != nil {
 		return fmt.Errorf("error finding matching public key for signature: %v", err)
 	}
+
 	now := time.Now()
-	var lastErr error
-	for _, accKey := range accKeys {
-		if accKey.isKeyValidAt(now) {
-			err := accKey.publicKey().verify(content, sig)
-			if err == nil {
-				// see if the assertion requires further checks
-				if checker, ok := assert.(consistencyChecker); ok {
-					err := checker.checkConsistency(db, accKey)
-					if err != nil {
-						return fmt.Errorf("signature verifies but assertion violates other knowledge: %v", err)
-					}
-				}
-				return nil
-			}
-			lastErr = err
-		}
+	if !accKey.isKeyValidAt(now) {
+		return fmt.Errorf("assertion is signed with expired public key by %q key id %q", assert.AuthorityID(), sig.KeyID())
```

Comment:
Similar to above, this message could be simplified, e.g. "assertion is signed with expired public key %q for %q".